### PR TITLE
refactor(ir): update `ArrayMap` to use the new `callable_with` validation rule

### PR DESCRIPTION
**Pull request by kszucs**
depends on https://github.com/ibis-project/ibis/pull/5515

### CODE REVIEW

File: ibis/expr/operations/arrays.py [hunk 1]
```diff
@@ -94,30 +91,28 @@ class ArrayRepeat(Value):
...
-        ]
-        expr = self.func(*args)
-        return expr.op()
-
-    @property
-    def signature(self):
-        return list(inspect.signature(self.func).parameters.keys())
+        arg = Argument(
+            name=self.parameter,
+            shape=rlz.Shape.COLUMNAR,
```
Comment (cpcloud) [inline ibis/expr/operations/arrays.py:105]: This isn't correct. The output shape depends on the input shape. Consider the case of a literal array.

**Review by cpcloud — COMMENTED** [review]
Looking pretty nice!

File: ibis/expr/operations/arrays.py (hunk 1 above)
Comment (kszucs) [reply ibis/expr/operations/arrays.py:105]: Updating.

**Comment by kszucs** [comment]
@cpcloud ready for another look

**Review by cpcloud — APPROVED** [review]

**Pull request merged**